### PR TITLE
Implement styles dialog when exporting style[s].

### PR DESCRIPTION
**Pull request by darkelectron**
This implements a dialog for overwriting existing style(s) when exporting. It uses a similar dialog to copy/move dialog from file managers.

Hey, @junkyardsparkle.

### CODE REVIEW

**Comment by junkyardsparkle** [comment]
I only did quick test, but seems to be working here, thanks! One thing: "Applying this option" should read "Apply this option".

**Comment by darkelectron** [comment]
Thanks, fixed now.

File: src/libs/styles.c [hunk 1]
```diff
@@ -370,13 +374,115 @@ static void export_clicked(GtkWidget *w, gpointer user_data)
...
+          char overwrite_str[256];
+
+          GtkWidget *dialog_overwrite_export = gtk_dialog_new_with_buttons("overwrite style?", GTK_WINDOW(win), GTK_DIALOG_DESTROY_WITH_PARENT,
+              _("cancel"), GTK_RESPONSE_CANCEL,
+              _("skip"), GTK_RESPONSE_NONE,
+              _("overwrite"), GTK_RESPONSE_ACCEPT, NULL);
+
+          // contents for dialog
+          GtkWidget *content_area = gtk_dialog_get_content_area(GTK_DIALOG(dialog_overwrite_export));
+          sprintf(overwrite_str, "style `%s' already exists.\ndo you want to overwrite existing style?\n", (char*)style->data);
```
Comment (TurboGit) [inline src/libs/styles.c:420]: this string must be translatable

File: src/libs/styles.c [hunk 2]
```diff
@@ -370,13 +374,115 @@ static void export_clicked(GtkWidget *w, gpointer user_data)
...
+          {
+            break;
+          }
+        }
+        else
+        {
+          /* create and run dialog */
+          char overwrite_str[256];
+
+          GtkWidget *dialog_overwrite_export = gtk_dialog_new_with_buttons("overwrite style?", GTK_WINDOW(win), GTK_DIALOG_DESTROY_WITH_PARENT,
```
Comment (TurboGit) [inline src/libs/styles.c:413]: string must be translatate

File: src/libs/styles.c [hunk 3]
```diff
@@ -370,13 +374,115 @@ static void export_clicked(GtkWidget *w, gpointer user_data)
...
+              _("cancel"), GTK_RESPONSE_CANCEL,
+              _("skip"), GTK_RESPONSE_NONE,
+              _("overwrite"), GTK_RESPONSE_ACCEPT, NULL);
+
+          // contents for dialog
+          GtkWidget *content_area = gtk_dialog_get_content_area(GTK_DIALOG(dialog_overwrite_export));
+          sprintf(overwrite_str, "style `%s' already exists.\ndo you want to overwrite existing style?\n", (char*)style->data);
+          GtkWidget *label = gtk_label_new(overwrite_str);
+          GtkWidget *overwrite_dialog_check_button = gtk_check_button_new_with_label("apply this option to all existing styles");
```
Comment (TurboGit) [inline src/libs/styles.c:422]: string should be translatable

**Review by TurboGit — CHANGES_REQUESTED** [review]
Some translations issues.

Also this dialog should be controlled by a security option. Maybe reusing the "ask before deleting a style" ?

Conf key is : `plugins/lighttable/style/ask_before_delete_style`

File: src/libs/styles.c [hunk 4]
```diff
@@ -536,7 +642,7 @@ void gui_init(dt_lib_module_t *self)
   /* filter entry */
   w = gtk_entry_new();
   d->entry = GTK_ENTRY(w);
-  gtk_entry_set_placeholder_text(GTK_ENTRY(d->entry), _("filter styles name"));
+  gtk_entry_set_placeholder_text(GTK_ENTRY(d->entry), _("filter style names"));
```
Comment (TurboGit) [inline src/libs/styles.c:645]: Also you want to rebase  your branch on top of current master as this change is already committed.

**Comment by darkelectron** [comment]
> Some translations issues.

For the translations, does that mean using `ngettext`.

**Comment by TurboGit** [comment]
> > Some translations issues.
> 
> For the translations, does that mean using `ngettext`.

No, just `_(" ... ")`

File: src/libs/styles.c [hunk 5]
```diff
@@ -370,13 +374,120 @@ static void export_clicked(GtkWidget *w, gpointer user_data)
...
+          {
+            GtkWidget *dialog_overwrite_export = gtk_dialog_new_with_buttons(_("overwrite style?"), GTK_WINDOW(win), GTK_DIALOG_DESTROY_WITH_PARENT,
+                _("cancel"), GTK_RESPONSE_CANCEL,
+                _("skip"), GTK_RESPONSE_NONE,
+                _("overwrite"), GTK_RESPONSE_ACCEPT, NULL);
+
+            // contents for dialog
+            GtkWidget *content_area = gtk_dialog_get_content_area(GTK_DIALOG(dialog_overwrite_export));
+            sprintf(overwrite_str, "style `%s' already exists.\ndo you want to overwrite existing style?\n", (char*)style->data);
+            GtkWidget *label = gtk_label_new(_(overwrite_str));
```
Comment (TurboGit) [inline src/libs/styles.c:426]: the `_(...)` should be above around the real string, not the variable.

**Review by TurboGit — CHANGES_REQUESTED** [review]
One more translation fix needed.

**Comment by TurboGit** [comment]
Looks good to me now. Thanks !

**Pull request merged**